pkg/plugin/windows/hnsstats/hnsstats_windows.go

```diff
@@ -96,34 +97,39 @@ func pullHnsStats(ctx context.Context, h *hnsstats) error {
 			if err != nil {
 				h.l.Error("Getting Vswitch ports failed", zap.Error(err))
 			}
+
 			for _, ep := range endpoints {
 				if len(ep.IpConfigurations) < 1 {
 					h.l.Info("Skipping endpoint without IPAddress", zap.String("EndpointID", ep.Id))
 					continue
 				}
-				var mac string = ep.MacAddress
-				var ip string = ep.IpConfigurations[0].IpAddress
-				// var id string = ep.Id
-				if stats, err := hcsshim.GetHNSEndpointStats(ep.Id); err != nil {
-					h.l.Error("Getting endpoint stats failed for endpoint ID "+ep.Id, zap.Error(err))
+
+				id := ep.Id
+				mac := ep.MacAddress
+				ip := ep.IpConfigurations[0].IpAddress
+
+				if stats, err := hcsshim.GetHNSEndpointStats(id); err != nil {
+					h.l.Error("Getting endpoint stats failed for endpoint ID "+id, zap.Error(err))
 				} else {
 					hnsStatsData := &HnsStatsData{hnscounters: stats, IPAddress: ip}
-					// h.l.Info(fmt.Sprintf("Fetched HNS endpoints stats for ID: %s, IP %s, MAC %s", id, ip, mac))
+					h.l.Info(fmt.Sprintf("Fetched HNS endpoints stats for ID: %s, IP %s, MAC %s", id, ip, mac))
```

Comment:
same here
```suggestion
					h.l.Info(fmt.Sprintf("Fetched HNS endpoints stats", zap.String("endpointID", id), zap.String("IP", ip), zap.String("MAC", mac))
```

Reply:
I was a little bit unsure about mixing Sprintf and zap logger. Thanks for your suggestion/help, I'm updating it!

Reply:
@rbtr I don't think this one would work, because of `Sprintf` (that expects a format and an interface). Do you think we could keep this one without the `zap` wrapper? There are some other occurrences of `Sprintf` without `zap` in this file:

![image](https://github.com/microsoft/retina/assets/48071009/9f2bcbb8-1715-4e63-86b3-db0730f3ebde)

Reply:
Oops, updated my comment - should have removed the `Sprintf`

Reply:
For those other occurrences, they are basically formatting the log around a key value pair. So we would want to replace it with `Debug("emitting metric", zap.String(key, val))`

Reply:
@rbtr indeed! give me a moment, gonna updated it too 😊 thanks for all of your help!